src/lsm/table_mutable.zig

```diff
@@ -174,10 +297,16 @@ pub fn TableMutableType(comptime Table: type, comptime tree_name: [:0]const u8)
             assert(values_max.len == value_count_max);
 
             var i: usize = 0;
-            var it = table.values.keyIterator();
+            var sorted = true;
+            var it = table.values.iterator();
             while (it.next()) |value| : (i += 1) {
                 values_max[i] = value.*;
 
+                if (i > 0 and sorted) {
+                    const prev_key = key_from_value(&values_max[i - 1]);
+                    sorted = compare_keys(prev_key, key_from_value(value)) != .gt;
+                }
```

Comment:
This is expensive to do in the data plane, since we're branching on `sorted` and comparing all keys.

The idea is that by this point, TableMutable should already know whether it needs to do any last minute sort finalization, or else if all keys are already sorted, and if so, it should be aiming to do a single memcpy here.

Reply:
The comparison of all keys to check if it's sorted (stops comparing when it's not sorted) either happens here or during `upsert/insert_no_clobber`. We can't do a memcpy as some Values may be deleted from `fetch_remove` and shouldn't get copied into `values_max` (the `active` bitset is used to detect this). It's also already traversed for updating the values_cache.

Reply:
> The comparison of all keys to check if it's sorted (stops comparing when it's not sorted) either happens here or during upsert/insert_no_clobber.

Yes, however, doing it during upsert/insert would be more incremental.

> We can't do a memcpy as some Values may be deleted from fetch_remove and shouldn't get copied into values_max (the active bitset is used to detect this).

I mean that we should *aim* to do a single memcpy, i.e. we can iterate the bits quickly to find the biggest memcpy we can do. Then skip. Then big memcpy etc.